src/test/java/org/dependencytrack/search/FuzzyVulnerableSoftwareSearchManangerTest.java

```diff
@@ -0,0 +1,75 @@
+package org.dependencytrack.search;
+
+import org.dependencytrack.model.Component;
+import org.dependencytrack.model.VulnerableSoftware;
+import org.dependencytrack.persistence.QueryManager;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import us.springett.parsers.cpe.Cpe;
+import us.springett.parsers.cpe.CpeParser;
+import us.springett.parsers.cpe.exceptions.CpeEncodingException;
+import us.springett.parsers.cpe.exceptions.CpeParsingException;
+import us.springett.parsers.cpe.exceptions.CpeValidationException;
+import us.springett.parsers.cpe.values.Part;
+
+import java.util.List;
+import java.util.regex.Pattern;
+
+import static org.mockito.Mockito.*;
+
+import static org.junit.Assert.*;
+
+public class FuzzyVulnerableSoftwareSearchManangerTest {
```

Comment:
It'd be great if we could have non-`@Ignore`d tests for this. Off the top of my head I don't know how we can construct and reset the global Lucene index, but there should be a way to do it. Maybe it's already done in some other test?

Reply:
I really don't know how to write the integration tests used by this project, I tried mocking stuff but had difficulty due to how Singletons are used.